[JIT] add support for torch.norm

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33818 [JIT] Add support for list()
* **#33783 [JIT] add support for torch.norm**

Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

Commit:
[JIT] add support for torch.norm

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"




[ghstack-poisoned]

vishwakftw commented on torch/functional.py:
```diff
@@ -803,39 +814,51 @@ def norm(input, p="fro", dim=None, keepdim=False, out=None, dtype=None):
-        elif dtype is None:
-            return torch._C._VariableFunctions.norm(input, p, dim, keepdim=keepdim, out=out)
-    return torch._C._VariableFunctions.norm(input, p, dim, keepdim=keepdim, dtype=dtype, out=out)
+            dim = [i for i in range(ndim)]  # noqa: C416 TODO: rewrite as list(range(m))
```
I noticed this in at least one other preceding PR as well, perhaps we could consider adding this as a separate function to reduce duplicate code.

eellison replied:
Sounds good. I’m also going to try to add list(range()) support to TS in the next week

vishwakftw commented on test/test_jit.py:
```diff
@@ -10221,6 +10221,12 @@ def cdist_fn():
 
+        def norm():
+            c = torch.tensor([[1, 2, 3], [-1, 1, 4]], dtype=torch.float)
+            return torch.norm(c, p="fro"), torch.norm(c, p="nuc"), torch.norm(c), torch.norm(c, p=.5)
```
Is it not necessary to test with different function arguments, or are they left out since they are covered by standard tests in test_torch

eellison replied:
Yea, I was mostly testing that it compiles with different types of arguments. Since test_torch tests with different argument values and the code is the same for python/TorchScript, that should be sufficient

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113


[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113


[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]

Commit:
Update on "[JIT] add support for torch.norm"


Fix for https://github.com/pytorch/pytorch/issues/20113

Differential Revision: [D20121917](https://our.internmc.facebook.com/intern/diff/D20121917)

[ghstack-poisoned]
